### PR TITLE
bump go

### PR DESCRIPTION
**Pull request by odsod**
- ci: bump Go to 1.16
- ci: bump GolangCI-Lint to 1.37.0

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -18,7 +18,12 @@ linters:
     - unparam # allow constant parameters
     - goerr113 # allow "dynamic" errors
     - dogsled # allow blank identifiers
-    - gocyclo # allow complex functions
+    - cyclop # allow complex functions
     - nlreturn # allow return/break without whitespace
     - exhaustive # TODO: make switches exhaustive and enable
     - noctx # TODO: update http calls and enable
+    - wrapcheck # TODO: wrap errors and enable
```
Comment (alethenorio) [inline .golangci.yml:25]: I'd just keep this off to be honest. Explicitly exposing every error without giving it some thought on each is a bad idea specially in an open source repo.

File: cmd/cantool/main.go [hunk 2]
```diff
@@ -163,7 +160,7 @@ func genGo(inputFile, outputFile string) error {
 	if err != nil {
 		return err
 	}
-	if err := ioutil.WriteFile(outputFile, output, 0600); err != nil {
+	if err := ioutil.WriteFile(outputFile, output, 0o600); err != nil {
```
Comment (alethenorio) [inline cmd/cantool/main.go:163]: Can be replaced with `os.Writefile` in 1.16

**Review by alethenorio — APPROVED** [review]
🙌

File: cmd/cantool/main.go (hunk 2 above)
Comment (alethenorio) [reply cmd/cantool/main.go:163]: Though to allow other running an older version probably best to keep it as it is

Comment (odsod) [reply cmd/cantool/main.go:163]: Let's give ourselves some time to transition over to 1.16 across the board, and then get at it! :smiley:

File: .golangci.yml (hunk 1 above)
Comment (odsod) [reply .golangci.yml:25]: true

**Pull request merged**